packages/builder/src/schemas.zod.ts

```diff
@@ -0,0 +1,304 @@
+import { z } from 'zod';
+import { handleZodErrors } from './error/zod';
+
+/// ================================ INPUT CONFIG SCHEMAS ================================ \\\
+
+export const contractSchema = z
+  .object({
+    artifact: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        create2: z.boolean(),
+        from: z.string(),
```

Comment:
could we have a zod custom validation for an address? the validation would just be `ethers.utils.isAddress(f)`. Definitely something that can be verified as part of syntax